ci: Limit cache size regardless of NO_DEPENDS

hebasto opened the pull request:
Close #18666.

hebasto committed:
ci: Limit cache size regardless of NO_DEPENDS

MarcoFalke commented:
ACK

Depends has ccache disabled anyway and is cached regardless of whether ccache is there or not, see #17248

MarcoFalke commented on ci/test/06_script_a.sh:
```diff
@@ -8,9 +8,7 @@ export LC_ALL=C.UTF-8
-if [ -z "$NO_DEPENDS" ]; then
-  DOCKER_EXEC ccache --max-size=$CCACHE_SIZE
-fi
+DOCKER_EXEC "command -v ccache > /dev/null && ccache --max-size=$CCACHE_SIZE"
```
```suggestion
DOCKER_EXEC "ccache --max-size=$CCACHE_SIZE"
```

Ccache is a requirement, why would this not be there?

hebasto replied:
> Ccache is a requirement for ci builds...

Yes, it is now. But it seems not guaranteed for custom defined `DOCKER_PACKAGES` macro:
https://github.com/bitcoin/bitcoin/blob/661e8df1b63b213d2d5b0d7cba0733869b508af9/ci/test/00_setup_env.sh#L56

hebasto replied:
Maybe it is worth to disable ability to provide custom `DOCKER_PACKAGES` value as we have the `PACKAGES` macro on that purpose, no?

MarcoFalke replied:
I still like the ability to provide a custom DOCKER_PACKAGES. `ccache` is required for the ci in the same way that a compiler or python is required. Without it, it doesn't run.

hebasto replied:
[Updated](https://github.com/bitcoin/bitcoin/pull/18667#issuecomment-614706768).

hebasto commented:
Updated 4f29a488f19a5eb7ba21f4a9cce5a5fe9a4d8d3d  -> 087e7a3b1dd97094792a286ee34c69b9c2f37fc2 ([pr18667.01](https://github.com/hebasto/bitcoin/commits/pr18667.01) -> [pr18667.02](https://github.com/hebasto/bitcoin/commits/pr18667.02), [diff](https://github.com/hebasto/bitcoin/compare/pr18667.01..pr18667.02)):

- addressed @MarcoFalke's [comment](https://github.com/bitcoin/bitcoin/pull/18667#discussion_r409593544):
> Ccache is a requirement for ci builds, why would this not be there?

hebasto commented:
Updated 087e7a3b1dd97094792a286ee34c69b9c2f37fc2  -> 0c6318788beaf1a31aeba5a21f3f8bb5c07cea6c ([pr18667.02](https://github.com/hebasto/bitcoin/commits/pr18667.02) -> [pr18667.03](https://github.com/hebasto/bitcoin/commits/pr18667.03), [diff](https://github.com/hebasto/bitcoin/compare/pr18667.02..pr18667.03)):

- addressed @MarcoFalke's [comment](https://github.com/bitcoin/bitcoin/pull/18667#discussion_r409619427):
> I still like the ability to provide a custom DOCKER_PACKAGES. `ccache` is required for the ci in the same way that a compiler or python is required. Without it, it doesn't run.
